boards: update nucleo144 boards to default config guidelines

erwango opened the pull request:
This change is a part of #9246 and fixes default configuration for nucelo board (144 pins versions)

erwango committed:
boards: nucleo_f429zi: Enable USB

Enable USB on nucleo_f249zi.
Update clock to allow precise USB clock.
Tested on samples/subsys/usb/cdc_acm

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: nucleo_f429zi: fix PWM instance

Fix desynch between dts and Kconfig settings

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: nucleo_f412zg: Fix default config

According to GH7151: boards: Move existing boards to
"default configuration guidelines"
Add SPI1 Arduino compatible config
Add I2C1 Arduino compatible config
Add USART6 Arduino compatible config
Disable Features by default.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

erwango committed:
boards: nucleo_f413zh: Fix default config

According to GH7151: boards: Move existing boards to
"default configuration guidelines"
Add SPI1 Arduino compatible config
Add I2C1 Arduino compatible config
Add USART6 Arduino compatible config
Disable Features by default.

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

dbkinder reviewed (approved):
doc changes LGTM, thanks.

erwango committed:
boards: nucleo_f207zg: Fix default config

According to GH7151: boards: Move existing boards to
"default configuration guidelines"
Add USART6 Arduino compatible config

Signed-off-by: Erwan Gouriou <erwan.gouriou@linaro.org>

MaureenHelm reviewed (commented):
Generally looks good, but please address the merge conflict.

erwango commented on boards/arm/nucleo_f413zh/nucleo_f413zh.yaml:
```diff
@@ -5,7 +5,12 @@ arch: arm
+ram: 320
+flash: 1536
 supported:
-  - usb_device
```
Rebase issue I guess. Let me fix this. Thanks!
